### ISSUE TITLE
i18nFormat throws ICU error when timezone is valid, but not proper case.

### ISSUE THREAD

**Issue by riddlemd** [cakephp/cakephp#12786]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.6.13
* Platform and Target: Ubuntu, Apache, MySql

### What you did
```
$time = new \Cake\I18n\Time();

$timezone = new \DateTimeZone('america/chicago');

echo $time->i18nFormat('yyyy-MM-dd'T'HH:mm:ss', $timezone);
```

### What happened
Providing improperly cased Timezone names (america/chicago vs America/Chicago) creates valid DateTimeZone, but causes ICU errors. 

Error: 

> Your version of icu does not support creating a date formatter for `.0.0.america/chicago.1.yyyy-MM-dd'T'HH:mm:ss`. You should try to upgrade libicu and the intl extension.

### What you expected to happen
I would expect ICU to use the timezone correctly as the information PHP provides IS correct, outside of capitalization.

**Comment by markstory** [comment]
Ok. The 'error' we get back from libicu is `false` and historically the most common reason that date formatter creation fails is because of old versions of libicu/intl that lack the requested locale information.

What do you think that error message should say instead?

**Comment by riddlemd** [comment]
Could we not do a quick check before throwing the current Exception and check if DateTimeZone::getName() is invalid and if so, throw a different Exception explaining that?

**Comment by markstory** [comment]
Yeah that could be done. The relevant code is https://github.com/cakephp/cakephp/blob/98d559c6071555bba26fdae8fc9c87acfb169eca/src/I18n/DateFormatTrait.php#L226-L232 if you come up with an error message that would help future you it will likely help others as well :smile:
